File: easy/443/string_compression.py

class Solution:
    def compress(self, chars: List[str]) -> int:
        N = len(chars)
        c = chars[0]
        i = 0
        k = 0

        while i < N:
            while i < N:
                if c == chars[i]:
                    k += 1
                    i += 1
                else:
                    c = chars[i]
                    break
            
            if k > 1:
                str_k = str(k)
                p = (i - k) + 1
                
                for s in str_k:
                    chars[p] = s
                    p += 1
                
                while p < i:
                    del chars[p]
                    i -= 1
                    N -= 1

            k = 0
        
        return len(chars)

File: easy/443/string_compression.cpp

#include <string>

class Solution {
public:
    int compress(vector<char>& chars) {
        int N, i, k, p;
        std::string::iterator s_k;
        string str_k;        
        char c;

        N = chars.size();
        c = chars[0];
        i = 0;
        k = 0;

        while (i < N) {
            while (i < N) {
                if (c == chars[i]) {
                    k++;
                    i++;
                }
                else {
                    c = chars[i];
                    break;
                }
            }
            
            if (k > 1) {
                str_k = std::to_string(k);
                s_k = str_k.begin();
                p = (i - k) + 1;
                
                while (s_k != str_k.end()) {
                    chars[p] = *s_k;
                    s_k++;
                    p++;
                }
                
                while (p < i) {
                    chars.erase(chars.begin() + p);
                    i--;
                    N--;
                }
            }
            
            k = 0;
        }
    
        return chars.size();
    }
};
